server/src/test/java/org/elasticsearch/cluster/routing/RoutingTableGenerator.java

```diff
@@ -129,7 +130,11 @@ public void update(ShardRouting shardRouting) {
             if (shardRouting.initializing()) {
                 initializing++;
             } else {
-                unassigned++;
+                if (shardRouting.primary()) {
+                    unassignedPrimary++;
+                } else {
+                    unassigned++;
+                }
```

Comment:
If I am not mistaken, here we changed the semantics of the `unassigned`. In the original version of the code it used to count all unassigned shards and now it is counting only the non-primaries. Was that intentional?